src/change_stream/options.rs

```diff
@@ -0,0 +1,79 @@
+//! Contains options for ChangeStreams.
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+use std::time::Duration;
+use typed_builder::TypedBuilder;
+
+use crate::{change_stream::event::ResumeToken, collation::Collation, options::AggregateOptions};
+
+/// These are the valid options that can be passed to the `watch` method for creating a
+/// [`ChangeStream`](crate::change_stream::ChangeStream).
+#[skip_serializing_none]
+#[derive(Clone, Debug, Default, Deserialize, Serialize, TypedBuilder)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct ChangeStreamOptions {
+    /// When set to [`FullDocumentType::UpdateLookup`], the
+    /// [`ChangeStreamEvent::full_document`](`crate::change_stream::event::ChangeStreamEvent::full_document`)
+    /// field  will be populated with a copy of the entire document that was updated from some time
+    /// after the change occurred when an "update" event occurs. By default, the `full_document`
+    /// field will be empty for updates.
+    #[builder(default)]
+    pub full_document: Option<FullDocumentType>,
+
+    /// Specifies the logical starting point for the new change stream. Note that if a watched
+    /// collection is dropped and recreated or newly renamed, `start_after` should be set instead.
+    /// `resume_after` and `start_after` cannot be set simultaneously.
+    ///
+    /// For more information on resuming a change stream see the documentation [here](https://docs.mongodb.com/manual/changeStreams/#change-stream-resume-after)
+    #[builder(default)]
+    pub resume_after: Option<ResumeToken>,
+
+    /// The maximum amount of time for the server to wait on new documents to satisfy a change
+    /// stream query.
+    #[builder(default)]
+    #[serde(skip_serializing)]
+    pub max_await_time: Option<Duration>,
+
+    /// The number of documents to return per batch.
+    #[builder(default)]
+    #[serde(skip_serializing)]
+    pub batch_size: Option<i32>,
+
+    /// Specifies a collation.
+    #[builder(default)]
+    #[serde(skip_serializing)]
+    pub collation: Option<Collation>,
+
+    /// The change stream will only provide changes that occurred at or after the specified
+    /// timestamp. Any command run against the server will return an operation time that can be
+    /// used here.
+    #[builder(default)]
+    pub start_at_operation_time: Option<i64>,
```

Comment:
This value should never be negative, right? If so we should use a `u64`. (Alternatively we could use the `bson::Timestamp` type here; the spec isn't super prescriptive and just indicates `Timestamp` for this type.)

Reply:
Updated to use `bson::Timestamp`.